### ISSUE TITLE
False negative YodaCondition

### ISSUE THREAD

**Issue by luke-hill** [rubocop/rubocop#12423]
My thoughts are this is a hard one to diagnose. YodaCondition should only trigger when it knows about a failure / issue. My issue distils down to me not actually performing a Yoda Condition. I'm asking if a version is greater than a number, but it auto--corrects it to whether a number is less than a version

--------

## Expected behavior

Rubocop returning 0 offenses

## Actual behavior

Rubocop returns YodaCondition error

```
lib/site_prism/dsl/locators.rb:75:60: C: [Correctable] Style/YodaCondition: Reverse the order of the operands Capybara::VERSION >= require_version.
        raise SitePrism::UnsupportedGemVersionError unless Capybara::VERSION >= require_version
                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

82 files inspected, 1 offense detected, 1 offense autocorrectable
```

## Steps to reproduce the problem

```
required_version = '3.37.0'
return true if Capybara::VERSION >= required_version
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
1.53.1 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-performance 1.19.0
  - rubocop-rspec 2.23.2
```

**Comment by koic** [comment]
This issue might be resolved with your configuration change. Perhaps setting `EnforcedStyle: forbid_for_equality_operators_only` would be the solution?

```yaml
# .rubocop.yml
Style/YodaCondition:
  EnforcedStyle: forbid_for_equality_operators_only
```

https://docs.rubocop.org/rubocop/1.57/cops_style.html#styleyodacondition

**Comment by koic** [comment]
In https://github.com/rubocop/rubocop/issues/11511, the `Style/YodaCondition` cop now treats constants like literals. Therefore, `Capybara::VERSION >= require_version` is not a comparison between two variables, but rather a comparison between a constant and a variable, which will be handled by the `Style/YodaCondition` cop.

**Comment by luke-hill** [comment]
Aaaah oki. So it's literally treating it as 3.37 >= variable. That makes a lot of sense. I'm not sure I fully agree with that statement, but at least you've explained the rationale.

Maybe when that's being compared to a literal it should be on LHS. Maybe this is an intractable problem?

**Comment by koic** [comment]
While I don't have a strong stance, considering that literals and constants are handled similarly, it seems possible to not prioritize one over the other. For instance, if the `Style/YodaCondition` cop doesn't register offenses for `1 > 2`, it could be considered that it shouldn't register offenses when one side of the literals is replaced with a constant.
